src/tests/src/solver/simulation/remix-storage/test-hydro-remix.cpp

```diff
@@ -49,19 +60,43 @@ struct InputFixture
                                  reservoirManagement);
     }
 
-    void callRemixStorageAlgorithm()
+    void callRemixStorageAlgorithm(unsigned nbStorage = 1)
     {
-        Load = TotalGenNoHydro + UnsupE + HydroGen;
+        Load = TotalGenNoHydro + UnsupE + HydroGen + HydroGen2 + HydroGen3;
         storagesForRemix.clear();
-        storagesForRemix.push_back(createHydroForRemix());
+        storagesForRemix.push_back(
+          createHydroForRemix(HydroGen, levels, inflows, init_level, capacity));
+        shavePeaksByRemixingStorageGen(Load, UnsupE, Spillage, DTG_MRG, storagesForRemix);
+
+        if (nbStorage >= 2)
+        {
+            auto storage2 = createHydroForRemix(HydroGen2,
+                                                levels2,
+                                                inflows2,
+                                                init_level2,
+                                                capacity2);
+            storagesForRemix.push_back(storage2);
+
+            if (nbStorage == 3)
+            {
+                auto storage3 = createHydroForRemix(HydroGen3,
+                                                    levels3,
+                                                    inflows3,
+                                                    init_level3,
+                                                    capacity3);
+                storagesForRemix.push_back(storage3);
+            }
```

Comment:
Can be improved, for example by creating a `struct` containing
- hydrogen
- levels
- inflows
- init level
- capacity

then `createHydroForRemix` takes this struct as it's only argument